packages/core/__tests__/bases/state.test.ts

```diff
@@ -0,0 +1,135 @@
+import { StateBase } from '../../src/bases/state';
+import { DappEnv, State, WalletStatus } from '../../src/types';
+
+describe('StateBase', () => {
+  let stateBase: StateBase;
+  beforeEach(() => {
+    stateBase = new StateBase();
+  });
+
+  it('should initialize with state "Init"', () => {
+    expect(stateBase.state).toBe('Init');
+  });
+
+  it('should set the state', () => {
+    stateBase.setState(State.Done);
+    expect(stateBase.state).toBe('Done');
+  });
+
+  it('should set the data', () => {
+    const data = { foo: 'bar' };
+    stateBase.setData(data);
+    expect(stateBase.data).toBe(data);
+  });
+
+  it('should set the message', () => {
+    const message = 'Hello, world!';
+    stateBase.setMessage(message);
+    expect(stateBase.message).toBe(message);
+  });
+
+  it('should reset the state, data, and message', () => {
+    stateBase.setState(State.Done);
+    stateBase.setData({ foo: 'bar' });
+    stateBase.setMessage('Hello, world!');
+    stateBase.reset();
+    expect(stateBase.state).toBe('Init');
+    expect(stateBase.data).toBeUndefined();
+    expect(stateBase.message).toBeUndefined();
+  });
+
+  it('should set the environment', () => {
+    const env = {
+      device: 'desktop',
+      os: 'macos',
+      browser: 'chrome',
+    };
+    stateBase.setEnv(env as DappEnv);
+    expect(stateBase.env).toBe(env);
+  });
+
+  it('should set the actions', () => {
+    const actions = { action1: jest.fn(), action2: jest.fn() };
+    stateBase.setActions(actions);
+    expect(stateBase.actions).toBe(actions);
+  });
+
+  it('should check if it is mobile', () => {
+    // Mock the isMobile function
+    jest.spyOn(stateBase, 'env', 'get').mockReturnValue({ device: 'mobile' });
+    expect(stateBase.isMobile).toBe(true);
+  });
+
+  it('should check if it is initialized', () => {
+    // Mock the isInit function
+    jest.spyOn(stateBase, 'state', 'get').mockReturnValue(State.Init);
+    expect(stateBase.isInit).toBe(true);
+  });
+
+  it('should check if it is done', () => {
```

Comment:
correct me if I'm wrong but why are we mocking the return value and then assert that?
Shouldn't we test the behavior, not the implementation?